
syntax = "proto2";
package managarm.kerncfg;
option optimize_for = LITE_RUNTIME;

enum Error {
	SUCCESS = 0;
	ILLEGAL_REQUEST = 1;
}

enum CntReqType {
	NONE = 0;
	GET_CMDLINE = 1;
	GET_BUFFER_CONTENTS = 2;
}

message CntRequest {
	optional CntReqType req_type = 1;
	optional uint64 size = 2;
	optional uint64 dequeue = 3;
}

message SvrResponse {
	optional Error error = 1;
	optional uint64 size = 2;
	optional uint64 new_dequeue = 3;
	optional uint64 enqueue = 4;
}

